### PR TITLE
fix: The app gets stuck on splash screen

### PR DESCRIPTION
**Pull request by mejdoo**
### Issues

The app gets stuck on splash screen

### Causes

There is an Assets2Module loaded in WireApplication class which is executed in a blockingIO thread. 

### Solutions

Changing all the IO operations to a separate threadPool instead of using a blocking IO thread

### Testing

We should reproduce the app stuck scenario and test it again with new fix

#### APK
[Download build #3](http://10.10.124.11:8080/job/Pull%20Request%20Builder/3/artifact/build/artifact/wire-dev-PR2417-3.apk)
[Download build #361](http://10.10.124.11:8080/job/Pull%20Request%20Builder/361/artifact/build/artifact/wire-dev-PR2417-361.apk)
[Download build #362](http://10.10.124.11:8080/job/Pull%20Request%20Builder/362/artifact/build/artifact/wire-dev-PR2417-362.apk)

### CODE REVIEW

**Comment by johnxnguyen** [comment]
build again

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/threading/Threading.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ object Threading {
     implicit lazy val Background: DispatchQueue    = Threading.ThreadPool
     implicit lazy val Ui:         DispatchQueue    = Threading.Ui
     implicit lazy val Image:      DispatchQueue    = Threading.ImageDispatcher
-    implicit lazy val BlockingIO: ExecutionContext = Threading.BlockingIO
+    implicit lazy val IO: ExecutionContext = Threading.IO
```
Comment (johnxnguyen) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/threading/Threading.scala]: This is the `IOThreadPool`, whose comment states `Thread pool for blocking IO tasks.`. Is this intended?

Comment (mejdoo) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/threading/Threading.scala]: Yes that's it IOThreadPool.

Comment (mejdoo) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/threading/Threading.scala]: For ProGuard, I don't think we need to update the files as we don't have new dependencies.

**Comment by gizemb** [comment]
build again

**Comment by marcoconti83** [comment]
build again

**Comment by marcoconti83** [comment]
build again

**Pull request merged**